e2e/gpu/gpu.go

```diff
@@ -95,11 +96,115 @@ func (c ctx) testNvidia(t *testing.T) {
 			e2e.WithProfile(e2e.UserProfile),
 			e2e.WithCommand("exec"),
 			e2e.WithArgs(tt.args...),
+			e2e.WithEnv(tt.env),
 			e2e.ExpectExit(0),
 		)
 	}
 }
 
+func (c ctx) testNvCCLI(t *testing.T) {
+	require.Nvidia(t)
+	require.NvCCLI(t)
+	// Use Ubuntu 20.04 as this is a recent distro officially supported by Nvidia CUDA.
+	// We can't use our test image as it's alpine based and we need a compatible glibc.
+	imageURL := "docker://ubuntu:20.04"
+	imageFile, err := fs.MakeTmpFile("", "test-nvccli-", 0o755)
+	if err != nil {
+		t.Fatalf("Could not create test file: %v", err)
+	}
+	imageFile.Close()
+	imagePath := imageFile.Name()
+	defer os.Remove(imagePath)
+
+	c.env.RunSingularity(
+		t,
+		e2e.WithProfile(e2e.UserProfile),
+		e2e.WithCommand("pull"),
+		e2e.WithArgs("--force", imagePath, imageURL),
+		e2e.ExpectExit(0),
+	)
+
+	// Basic test that we can run the bound in `nvidia-smi` which *should* be on the PATH
+	tests := []struct {
+		name        string
+		profile     e2e.Profile
+		args        []string
+		env         []string
+		expectExit  int
+		expectMatch e2e.SingularityCmdResultOp
+	}{
+		{
+			name:       "User",
+			profile:    e2e.UserProfile,
```

Comment:
I was wondering why those tests are passing with setuid workflow, but they are just skipped because nvidia-container-cli/nvidia-smi are not installed on CICD host, sylabs didn't run those tests in their CICD pipeline either. But for dev running those tests on their machine with this PR, it would fail, the behavior need to be reflected in those tests

Reply:
If the tests are run with 'sudo', then they should still pass, right?  Are these tests run with the setuid flow?

Reply:
Oh I guess "e2e.UserProfile" means setuid.  Should I change it to UserNamespaceProfile or RootProfile?  Which would be better?

Reply:
I changed them to RootProfile because there was one test at the end already for UserNamespaceProfile.  There was one also for RootProfile and I removed that.  I also noticed that the per-test profile was being ignored in that new test function and two old ones in the file, always setting them to UserProfile, and I fixed those.